src/ws/test-kerberos.c

```diff
@@ -403,9 +403,11 @@ mock_kdc_start (void)
       ret = read (out_fd, input->str + len, 256);
       if (ret < 0)
         {
-          if (errno != EAGAIN && errno != EINTR)
-            g_error ("couldn't read from mock-kdc: %s", g_strerror (errno));
-            break;
+          if (errno == EAGAIN || errno == EINTR)
+            continue;
```

Comment:
I don't like this restructuring.  Please either keep the original structure (with added braces), or do your approach, but then remove the non-error case from the `else` (and unindent it).  There's no sense in having continue; and break; before you effectively hit the end of the body of the loop (because you skip the else)...

Reply:
The original structure (with added braces) made no sense. It would terminate the read loop on EAGAIN/EINTR, which is exactly what these error codes *don't* as you to do.

Indeed the `break` is redundant as `g_error()` is fatal, but I suppose that's for the sake of readability or so? But indeed I agree to dropping the else, that's easier to read.

Reply:
Looks nicer, thanks.

The further handling of the `ret == 0` case below is a bit off-putting, but let's not go too wild here...